### PR TITLE
feat(servicelevel): update code gen strategy

### PR DESCRIPTION
**Pull request by ctrombley**

### CODE REVIEW

**Review by sanderblue — DISMISSED** [review]

**Review by asanuy — COMMENTED** [review]
Good job @ctrombley, I really appreciate that you took the time to take a look at this! :)

Just a few comments regarding some references to the `slug` that still remain in the code.

I can see that the mutations still hold references to the field, for instance:

https://github.com/newrelic/newrelic-client-go/blob/main/pkg/servicelevel/servicelevel_api.go#L108

Checking the configuration in the tutone library, I seems that it would be possible to exclude fields in mutations using the [excludeFields](https://github.com/newrelic/tutone/blob/main/internal/config/config.go#L118) property. Do you think it would work?

Besides that, I can see that in the `pkg/entities` there are also some references to the field. I guess we could apply similar exceptions as in the `servicelevel` package?

For the records, these are the references in the `pkg/entities`:

https://github.com/newrelic/newrelic-client-go/blob/main/pkg/entities/types.go#L6602
https://github.com/newrelic/newrelic-client-go/blob/main/pkg/entities/entities_api.go#L344
https://github.com/newrelic/newrelic-client-go/blob/main/pkg/entities/entities_api.go#L1141

**Review by asanuy — DISMISSED** [review]
Brilliant, looking sharp now. Thanks for taking care of it! 🍏

**Review by asanuy — APPROVED** [review]

**Review by jthurman42 — APPROVED** [review]
:shipit:

File: pkg/servicelevel/types.go [hunk 1]
```diff
@@ -251,6 +246,9 @@ type UserReference struct {
 	Name string `json:"name,omitempty"`
 }
 
+// EntityGUID - An encoded Entity GUID
+type EntityGUID string
```
Comment (jthurman42) [inline pkg/servicelevel/types.go:250]: We should think about moving this out to a very basic module that has no dependancies to solve these circular import issues for common types and not re-defining the type.

**Pull request merged**